src/root.component.tsx

```diff
@@ -1,26 +1,42 @@
 import React from "react";
+import openmrsRootDecorator from "@openmrs/react-root-decorator";
+import styles from "./root.styles.css";
 
-export default class Root extends React.Component {
-  state = {
-    catastrophicError: false
-  };
-  render() {
-    return this.state.catastrophicError
-      ? this.errorHasOccurred()
-      : this.loadNavigation();
-  }
-  componentDidCatch() {
-    this.setState({ catastrophicError: true });
-  }
-  errorHasOccurred = () => {
-    // TO-DO have a good UX for catastrophic errors
-    return null;
-  };
-  loadNavigation = () => {
-    return (
+function Root() {
+  const [sidenavOpen, setSidenavOpen] = React.useState(false);
+
+  React.useEffect(() => {
+    const operation = sidenavOpen ? "add" : "remove";
+    document.body.classList[operation]("omrs-sidenav-expanded");
```

Comment:
See related https://github.com/openmrs/openmrs-esm-styleguide/pull/26 for the css transitions that happen to main content elements because of the `omrs-sidenav-expanded` css class.

Also see https://developer.mozilla.org/en-US/docs/Web/API/Element/classList for more info on what we're doing with `classList`